src/core/thread/network_data_service.cpp

```diff
@@ -0,0 +1,187 @@
+                                              reinterpret_cast<const uint8_t *>(aServerData), aServerDataLength));
+
+    Get<Notifier>().HandleServerDataUpdated();
+
+exit:
+    return error;
+}
+
+otError Manager::RemoveService(uint8_t aServiceNumber)
+{
+    otError error;
+
+    SuccessOrExit(error = Get<Local>().RemoveService(kThreadEnterpriseNumber, &aServiceNumber, sizeof(uint8_t)));
+    Get<Notifier>().HandleServerDataUpdated();
+
+exit:
+    return error;
+}
+
+#endif // OPENTHREAD_CONFIG_TMF_NETDATA_SERVICE_ENABLE
+
+otError Manager::GetServiceId(uint8_t aServiceNumber, bool aServerStable, uint8_t &aServiceId) const
+{
+    return Get<Leader>().GetServiceId(kThreadEnterpriseNumber, &aServiceNumber, sizeof(uint8_t), aServerStable,
+                                      aServiceId);
+}
+
+#if (OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2)
+
+otError Manager::GetBackboneRouterPrimary(ot::BackboneRouter::BackboneRouterConfig &aConfig) const
+{
+    otError                           error          = OT_ERROR_NOT_FOUND;
+    const uint8_t                     serviceData    = BackboneRouter::kServiceNumber;
+    const ServerTlv *                 rvalServerTlv  = nullptr;
+    const BackboneRouter::ServerData *rvalServerData = nullptr;
+    Iterator                          iterator;
+
+    iterator.mServiceTlv = Get<Leader>().FindService(kThreadEnterpriseNumber, &serviceData, sizeof(serviceData));
+
+    VerifyOrExit(iterator.mServiceTlv != nullptr, aConfig.mServer16 = Mac::kShortAddrInvalid);
+
+    while (IterateToNextServer(iterator) == OT_ERROR_NONE)
+    {
+        const BackboneRouter::ServerData *serverData;
+
+        if (iterator.mServerSubTlv->GetServerDataLength() < sizeof(BackboneRouter::ServerData))
+        {
+            continue;
+        }
+
+        serverData = reinterpret_cast<const BackboneRouter::ServerData *>(iterator.mServerSubTlv->GetServerData());
+
+        if (rvalServerTlv == nullptr ||
+            (iterator.mServerSubTlv->GetServer16() ==
+             Mle::Mle::Rloc16FromRouterId(Get<Mle::MleRouter>().GetLeaderId())) ||
+            serverData->GetSequenceNumber() > rvalServerData->GetSequenceNumber() ||
+            (serverData->GetSequenceNumber() == rvalServerData->GetSequenceNumber() &&
+             iterator.mServerSubTlv->GetServer16() > rvalServerTlv->GetServer16()))
+        {
+            rvalServerTlv  = iterator.mServerSubTlv;
+            rvalServerData = serverData;
+        }
+    }
+
+    VerifyOrExit(rvalServerTlv != nullptr);
+
+    aConfig.mServer16            = rvalServerTlv->GetServer16();
+    aConfig.mSequenceNumber      = rvalServerData->GetSequenceNumber();
+    aConfig.mReregistrationDelay = rvalServerData->GetReregistrationDelay();
+    aConfig.mMlrTimeout          = rvalServerData->GetMlrTimeout();
+
+    error = OT_ERROR_NONE;
+
+exit:
+    return error;
+}
+
+#endif // (OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2)
+
```

Comment:
add macro `#if OPENTHREAD_CONFIG_SRP_SERVER_ENABLE || OPENTHREAD_CONFIG_SRP_CLIENT_ENABLE` for `GetNextSrpServerInfo` method?

Reply:
I actually had such an `#if` check but removed it.

The reason is I think it is good to have these APIs/feature to discover list of server from network data independent of SRP client/server features (even if the OT SRP client and/or server are not being used).

For example,  a system/product may have its own SRP client implementation (part of its own sw stack) and not want to use the included SRP client from OT but it can be helpful for it to be able to discover the SRP servers (from network data).